text/0224-experimental-semantic-conventions.md

```diff
@@ -0,0 +1,115 @@
+# Allowing experimental extensions to Semantic Conventions
```

Comment:
Perhaps `experimental attributes` would be more precise? 
Conventions describe relationships between spans, how attributes depend on other attributes, span kind, name, events populated under spans, context. etc. It'd be hard to apply process outlined here for things like that.

Reply:
An example: 
messaging WG discusses that attributes can be populated on spans or on links. E.g. span that represents batch publish should have `msg.batch.message_count` attribute AND links to all individual messages. This is equally important to the backend as an attribute name.

You can't standardize and regulate attributes without also regulating the stability of the behavior. But I don't think this document addresses the evolution of behavior. 

Hence, I think this document should be limited to attributes only and don't attempt to tackle semantic convention as a whole

Reply:
> I think this document should be limited to attributes only and don't attempt to tackle semantic convention as a whole

I like this idea, something like

```suggestion
# Supporting experimental attributes in OpenTelemetry instrumentation
```

Reply:
@jsuereth what do you think about this suggestion? it is a bit of a change in scope but does more closely match what we've laid out.

Reply:
I limited the scope of the proposal to just an attribute dictionary specifically to address this concern. If it goes well, maybe the idea will be extended to other areas of semantic conventions in the future.